glean-core/ffi/src/lib.rs

```diff
@@ -50,6 +54,28 @@ unsafe fn from_raw_string_array(arr: RawStringArray, len: i32) -> Vec<String> {
         .collect()
 }
 
+/// Create a HashMap<T, String> from a pair of C int and string arrays.
+unsafe fn from_raw_int_array_and_string_array(
+    keys: RawIntArray,
+    values: RawStringArray,
+    len: i32,
+) -> Option<HashMap<i32, String>> {
+    if keys.is_null() || values.is_null() || len == 0 {
+        return None;
+    }
+
+    let keys_ptrs = std::slice::from_raw_parts(keys, len as usize);
+    let values_ptrs = std::slice::from_raw_parts(values, len as usize);
+
+    Some(
+        keys_ptrs
+            .iter()
+            .zip(values_ptrs.iter())
+            .map(|(&k, &v)| (k, FfiStr::from_raw(v).into_string()))
```

Comment:
This can potentially fail, see #179, but as mentioned above let's fix that in a follow up